features/vaadin-dashlets/dashlet-ksc/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/KscDashlet.java

```diff
@@ -430,6 +425,19 @@ public Component getComponent() {
         });
     }
 
+    String determineNodeIdByResourceId(String resourceId){
+
+        // old implementation:
+        String[] arr0 = resourceId.split("\\.");
```

Comment:
I am not sure under which circumstances this code has worked previously but it doesn't seem to work anymore.

Reply:
When this dashlet was originally written the nodeId and not foureignSource:foreignId was reported inside the square brackets. This behavior has changed and did break these lines of code.

Reply:
Thank you @christianpape, I'll change the code